tasks/Task.cpp

```diff
@@ -126,17 +125,16 @@ bool Task::startHook()
     thread = new std::thread([this]{ this->server->loop(); });
 
     return true;
-    return true;
 }
 void Task::updateHook()
 {
     TaskBase::updateHook();
+    _received.write(received);
```

Comment:
Make one __timestamped__ statistics structure and output it. Also, would IMO be better to output it in the callback for the handler (updateHook will generally not be called in this component)

Reply:
> (updateHook will generally not be called in this component)
Why not? I put the task to be periodic. Being periodic the updateHook is called for each period don't? Also, The statics being periodic is good to know if doesn't have data due to errors or is not receiving messages.

>  structure and output it
So, some code will need to read it and could be another component. Thus, isn't it better to keep all these structures in a Rock Library?

Reply:
> So, some code will need to read it and could be another component. Thus, isn't it better to keep all these structures in a Rock Library?

All these structures are specific to this particular component. So, using them will be useful only in relation to this particular component and can depend on this package.

In practice, the stats will be read mostly during log postprocessing or by Syskit.

*If* the case arises that the type is more generic that we think, we'll move it to a common package the same way we did for `gps_base` or `ais_base`

Reply:
> Why not? I put the task to be periodic. Being periodic the updateHook is called for each period don't? Also, The statics being periodic is good to know if doesn't have data due to errors or is not receiving messages.

What "errors" ?